### PR TITLE
Implement HttpChaos

### PR DESCRIPTION
**Pull request by Hexilee**
Implementing HttpChaos through reconciler.

Signed-off-by: hexilee <i@hexilee.me>

### CODE REVIEW

File: api/v1alpha1/zz_generated.workflow.chaosmesh.go [hunk 1]
```diff
@@ -13,25 +13,27 @@
...
+
 const (
-	TypeAwsChaos     TemplateType = "AwsChaos"
-	TypeDNSChaos     TemplateType = "DNSChaos"
-	TypeGcpChaos     TemplateType = "GcpChaos"
-	TypeHTTPChaos    TemplateType = "HTTPChaos"
-	TypeIoChaos      TemplateType = "IoChaos"
-	TypeJVMChaos     TemplateType = "JVMChaos"
-	TypeKernelChaos  TemplateType = "KernelChaos"
+	TypeAwsChaos TemplateType = "AwsChaos"
```
Comment (AsterNighT) [inline api/v1alpha1/zz_generated.workflow.chaosmesh.go:25]: I notice that the indent frequently modified. Maybe you should `make check` or something before commit this file?

**Review by AsterNighT — COMMENTED** [review]
Rest LGTM. Any ideas on testing for now? I'm struggling on it.

File: api/v1alpha1/zz_generated.workflow.chaosmesh.go (hunk 1 above)
Comment (Hexilee) [reply api/v1alpha1/zz_generated.workflow.chaosmesh.go:25]: The `make check` command will fail currently, and I have executed `make fmt` command. Moreover, I think we should ignore generated file in any format tasks.

Comment (Hexilee) [reply api/v1alpha1/zz_generated.workflow.chaosmesh.go:25]: Maybe we should change the Makefile.

**Comment by Hexilee** [comment]
> 
> 
> Rest LGTM. Any ideas on testing for now? I'm struggling on it.

I plan to write some e2e test, but I just find a bug, which should be fixed first.

**Comment by AsterNighT** [comment]
> > Rest LGTM. Any ideas on testing for now? I'm struggling on it.
> 
> I plan to write some e2e test, but I just find a bug, which should be fixed first.

I cannot build nothing on `nirvana` successfully :facepalm:.

**Comment by Hexilee** [comment]
> 
> 
> > > Rest LGTM. Any ideas on testing for now? I'm struggling on it.
> > 
> > 
> > I plan to write some e2e test, but I just find a bug, which should be fixed first.
> 
> I cannot build nothing on `nirvana` successfully 🤦.

We can build image-e2e-helper, image-chaos-mesh and image-chaos-daemon successfully.

**Comment by YangKeao** [comment]
1. Please reduce the default log level of `tproxy`. The `TRACE` level is too noisy.
2. It seems that the `Recover` e2e test is not stable. Why?

File: api/v1alpha1/zz_generated.workflow.chaosmesh.go (hunk 1 above)
Comment (Hexilee) [reply api/v1alpha1/zz_generated.workflow.chaosmesh.go:25]: Ignored

**Comment by YangKeao** [comment]
As it has passed the test, I will merge this PR :+1: 

:beers: HTTPChaos finally comes to Chaos Mesh.

**Comment by agilob** [comment]
Any ETA for a stable release which includes HTTPChaos?

**Comment by Hexilee** [comment]
> 
> 
> Any ETA for a stable release which includes HTTPChaos?

Actually, the v2.0.0 includes HTTPChaos, but there are no dashboard supports yet, you can create the experiment by YAML config. The docs of v0.2.0 are going to release, you can preview them [here](https://chaos-mesh.org/docs/next/simulate-http-chaos-on-kubernetes).

**Comment by Hexilee** [comment]
> 
> 
> > As it has passed the test, I will merge this PR 👍
> > 🍻 HTTPChaos finally comes to Chaos Mesh.
> 
> @YangKeao Thanks for bringing this to life! It'll be a great addition to Chaos-Mesh.
> Curious to understand how this whole thing works? Basically which component in K8s cluster will honor the config and does the response transformation.

Hi, you can refer to the [rs-tproxy](https://github.com/chaos-mesh/rs-tproxy) repo to learn about how things work.

**Comment by agilob** [comment]
Thanks, I missed 2.0.0 release. I'm planning to use yaml experiments only, so that's ok!

**Pull request merged**